tests/integration/publish/app/test_cli_integ.py

```diff
@@ -0,0 +1,110 @@
+import os
+import time
+import json
+from subprocess import Popen, PIPE
+from unittest import TestCase
+import boto3
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+
+class TestPublishApp(TestCase):
+
+    def setUp(self):
+        self.region_name = "us-east-1"
+        self.sar_client = boto3.client('serverlessrepo', region_name=self.region_name)
+        self.test_data_path = Path(__file__).resolve().parents[2].joinpath("testdata", "publish")
+
+        # Create application for each test
+        app_metadata_text = self.test_data_path.joinpath("metadata_create_app.json").read_text()
+        app_metadata = json.loads(app_metadata_text)
+        app_metadata['TemplateBody'] = self.test_data_path.joinpath("template_create_app.yaml").read_text()
+        response = self.sar_client.create_application(**app_metadata)
+        self.application_id = response['ApplicationId']
+
+        # Avoid race conditions
+        time.sleep(1)
```

Comment:
I ran into intermittent errors that the application already exists because create_application is called immediately after delete_application, and another conflict exception would happen when I call update_application immediately after create_application.

Reply:
Are you using the same application name over and over again? I'd suggest adding a UUID to the application name for each test run so the data for each test is isolated. Also, I'm not a fan of sleeps in integ tests, and an arbitrary sleep time won't close the race condition completely. If they're happening to the tests, can't they happen to our customers too? Can we make the logic of the command (or the test if it's test-specific) robust enough that it can handle conflict errors?

Reply:
Thanks! The sleep logic is test-specific, it's not needed it anymore if I create a UUID application name each time.